### ISSUE TITLE
sana-sprint training issues about weighting_scheme

### ISSUE THREAD

**Issue by yizenghan** [NVlabs/Sana#236]
Hi,

I noticed "weighting_scheme: logit_normal_trigflow" is given in the training script configs/sana_sprint_config/1024ms/SanaSprint_1600M_1024px_allqknorm_bf16_scm_ladd.yaml.

In the training code train_scripts/train_scm_ladd.py, we see

                if weighting_scheme == "logit_normal_trigflow":
                    u = compute_density_for_timestep_sampling(
                        weighting_scheme=weighting_scheme,
                        batch_size=bs,
                        logit_mean=logit_mean,
                        logit_std=logit_std,
                        mode_scale=None,
                   )
                   denoise_timesteps = None

However, in diffusers, compute_density_for_timestep_sampling does not accept a weighting scheme of "weighting_scheme: logit_normal_trigflow": https://github.com/huggingface/diffusers/blob/main/src/diffusers/training_utils.py.

BTW, should we really use the weighting scheme "logit_normal_trigflow" instead of "logit_normal_trigflow_ladd"?

**Comment by lawrence-cj** [comment]
I don't see the problem about what we [have](https://github.com/NVlabs/Sana/blob/daa661fd85ffa881983d760006f3ea8598058c1f/diffusion/model/respace.py#L80) but diffusers not. 

This u you showed here is nothing about LADD. No need for the suffix '_ladd'.

**Comment by yizenghan** [comment]
Thanks for your detailed explanation!

**Comment by yizenghan** [comment]
Hi there, I've found another issue about weighting_scheme. We can see in the train file that config.scheduler.weighting_scheme_discriminator is required (https://github.com/NVlabs/Sana/blob/daa661fd85ffa881983d760006f3ea8598058c1f/train_scripts/train_scm_ladd.py). 

However, I did not found weighting_scheme_discriminator in the given config file (https://github.com/NVlabs/Sana/blob/daa661fd85ffa881983d760006f3ea8598058c1f/configs/sana_sprint_config/1024ms/SanaSprint_1600M_1024px_allqknorm_bf16_scm_ladd.yaml). Is it the same with the given weighting_scheme (logit_normal_trigflow)?

**Comment by yizenghan** [comment]
Another small issue: 

It seems that config.train.diff_timesteps_D is not provided in the config file.

**Comment by lawrence-cj** [comment]
We have the all of them here in the config.py file:
https://github.com/NVlabs/Sana/blob/daa661fd85ffa881983d760006f3ea8598058c1f/diffusion/utils/config.py#L125

**Comment by yizenghan** [comment]
Thanks! It seems that all missing values in SanaSprint_1600M_1024px_allqknorm_bf16_scm_ladd.yaml are set by default as in [Sana/diffusion/utils/config.py](https://github.com/NVlabs/Sana/blob/daa661fd85ffa881983d760006f3ea8598058c1f/diffusion/utils/config.py#L125). Thanks a lot!
